src/Compilers/Core/Portable/ReferenceManager/AssemblyData.cs

```diff
@@ -69,6 +70,17 @@ internal abstract class AssemblyData
             public abstract Compilation? SourceCompilation { get; }
 
             private string GetDebuggerDisplay() => $"{GetType().Name}: [{Identity.GetDisplayName()}]";
+#if DEBUG
+            public sealed override bool Equals(object? obj)
+            {
+                return base.Equals(obj);
+            }
+
+            public sealed override int GetHashCode()
+            {
+                return base.GetHashCode();
+            }
+#endif
         }
     }
```

Comment:
Please restore the code, it is there for a reason. ```MultiDictionary``` stores only unique values. We don't want any non-trivial code to be implicitly involved into the process of working with the ```MultiDictionary``` that use to bind/resolve references.

Reply:
Feel free to add a comment.

Reply:
Added back. I'll defer to you to add the comment at your convenience as you understand the intent better than I do.

Reply:
> I'll defer to you to add the comment at your convenience as you understand the intent better than I do.

Sounds good.